### ISSUE TITLE
scanning interfaces for matching ip/mask does not work of there are multiple IP addrs on an interface

### ISSUE THREAD

**Issue by AlpinChisholm** [alandau/arpspoof#18]
I have an interface servicing both 10.40.0.199 and 192.168.1.199
I was trying to find a target at 192.168.1.50 but the scan only checks the first address (10.40...) so it did not know this interface would work for that address. 
This is not trivial to fix since iface_info only supports one address at the moment.

**Comment by alandau** [comment]
Yeah, that's likely a complicated issue. I didn't even know Windows supports multiple IPs on the same interface. I'll see what can be done.

**Comment by AlpinChisholm** [comment]
Alex,
Here is some code I created to deal with the issue. Hopefully you will find
it helpful. (It still has a few debugs scattered around in it).
BTW I actually had to do an ARPSPOOF implementation and your sample code was
very helpful.
Al
 


//////////////////////////////////////////////////////////////////////
// Create a PCAP / pcap_t interface
// (pass a target IP address so ws can scan the interfaces for one that
services that IP address) //
//////////////////////////////////////////////////////////////////////
bool setup_pcap(pcap_t** pif, ULONG target) {
  char* errbuf = nullptr;

  pcap_if_t* select = NULL;
  pcap_if_t* pct = NULL;
  pcap_if_t* next = NULL;
  pcap_findalldevs(&pct, errbuf);
  select = next = pct;

  while (next)
  {
    pcap_addr* paddr = next->addresses;
    while (paddr)
    {
      printf("Fam:%2d, Name:%s, Desc:%s\n",
next->addresses->addr->sa_family, next->name, next->description);

      struct sockaddr_in* a;
      struct sockaddr_in* mask;

      a = (sockaddr_in*)paddr->addr;
      mask = (sockaddr_in*)paddr->netmask;

      char buf[64] = "";
      if (a && (a->sin_family == AF_INET))
      {
        TCPAddrToString(a->sin_addr, buf);
        printf("  Address:%s\n", buf);
        TCPAddrToString(mask->sin_addr, buf);
        printf("  MASK:%s\n", buf);

        ULONG nMask = mask->sin_addr.s_addr;
        ULONG nAddr = a->sin_addr.s_addr;
        if ((nAddr & nMask) == (target & nMask))
        {
          printf("Interface matches Target\n");
          select = next;
        }
      }
      paddr = paddr->next;
    }


    next = next->next;
  }

  char* dev = NULL;
  if (select) dev = select->name;
  //  dev = pcap_lookupdev(errbuf);
  if (dev == nullptr)
  {
    //LOG_F(ERROR, "Failed to read from network card");
    return false;
  }

  // Open the device for live capture, as opposed to reading a packet
  // capture file.
  //
  *pif = pcap_create(dev, errbuf);
  if (pif == nullptr)
  {
    //LOG_F(ERROR, "Failed to create pcap pif");
    pcap_freealldevs(pct);
    return false;
  }

  // Set the defaults
  //
  pcap_set_snaplen(*pif, 5000);   // Set the snapshot length to maximum UDP
size
  pcap_set_promisc(*pif, 1);        // Turn promiscuous mode on
  pcap_set_timeout(*pif, 512);      // Set the timeout to 512 milliseconds

  // Activate the device
  //
  const int status = ::pcap_activate(*pif);
  if (status < 0x0)
  {
    //LOG_F(ERROR, "Failed to create pcap pif");
    pcap_freealldevs(pct);
    return false;
  }

  // Done
  //

  pcap_freealldevs(pct);

  return status == 0;
}

 
 

________________________________
